src/main/java/com/chopshop166/chopshoplib/sensors/InvertDigitalInput.java

```diff
@@ -0,0 +1,50 @@
+package com.chopshop166.chopshoplib.sensors;
+
+import java.util.function.BooleanSupplier;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+
+/**
+ * uses DigitalInputSource interface to create a invertable input source (cuz !
+ * is boring)
+ */
+public class InvertDigitalInput extends DigitalInput implements DigitalInputSource {
+
+/**
+ * creates inverted boolean to be used for logic later
+ */
+     * 
+     private boolean inverted;
+
+/**
+ * allows input to change the inverted variable
+ */
+    public void setInverted(boolean invertValue) {
+        inverted = invertValue;
+    }
+
+    /**
+     * allows to check for inversion
+     */
+    public boolean getInverted() {
+        return inverted;
+    }
+
+    /**
+     * creates our input via the source interface
+     */
+    public InvertDigitalInput(int channel) {
```

Comment:
Constructors should go just after member variables and before member functions.

Reply:
🗸?